### PR TITLE
Rating calculation

### PR DESCRIPTION
**Pull request by TarasDovganyuk**
Created aspect where User rating is calculating

### CODE REVIEW

**Review by Yura09 — APPROVED** [review]

**Review by MarDatsko — APPROVED** [review]

**Comment by antonbakalets** [comment]
As discussed:

- lets replace method names with an enum.
- lets receive current user from SecurityContextHolder, not by parsing arguments.
- lets introduce a service that can be reused by admin part.

File: core/src/main/java/greencity/service/impl/PlaceCommentServiceImpl.java [hunk 1]
```diff
@@ -83,8 +86,9 @@ public CommentReturnDto save(Long placeId, AddCommentDto addCommentDto, String e
      *
      * @author Marian Milian
      */
+    @RatingCalculation(rating = RatingCalculationEnum.DELETE_COMMENT)
     @Override
-    public void deleteById(Long id) {
+    public void deleteById(Long id, User user) {
```
Comment (antonbakalets) [inline core/src/main/java/greencity/service/impl/PlaceCommentServiceImpl.java:91]: You don't need this User parameter anymore, did you?

File: core/src/main/java/greencity/service/impl/PlaceCommentServiceImpl.java [hunk 2]
```diff
@@ -93,13 +97,13 @@ public void deleteById(Long id) {
      * {@inheritDoc}
      */
     @Override
-    public PageableDto getAllComments(Pageable pageable) {
+    public PageableDto<CommentAdminDto> getAllComments(Pageable pageable) {
```
Comment (antonbakalets) [inline core/src/main/java/greencity/service/impl/PlaceCommentServiceImpl.java:100]: Like.

File: core/src/main/java/greencity/annotations/RatingCalculationEnum.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package greencity.annotations;
+
+import lombok.Getter;
+
+@Getter
+public enum RatingCalculationEnum {
```
Comment (antonbakalets) [inline core/src/main/java/greencity/annotations/RatingCalculationEnum.java:6]: There is no need to add `Enum` to the name, the same way as we don't add `Class` to the class name or `I` to interface name.

**Review by antonbakalets — APPROVED** [review]

**Pull request merged**